tests/js/common/shell/shell-document-babies.js

```diff
@@ -305,30 +306,43 @@ function CollectionDocumentSuiteBabies() {
     /// @brief insert with bad key types
     ////////////////////////////////////////////////////////////////////////////////
 
-    testInsertErrorBadKey: function() {
+    testInsertErrorBadKey: function () {
       var l = [null, false, true, 1, -1, {},
         []
       ];
-      l.forEach(function(k) {
-        var docs = collection.insert([{
+      l.forEach(function (k) {
+        const batch = [{
           _key: "a"
         }, {
           _key: k
-        }]);
-        assertEqual(docs.length, 2);
-        assertEqual(docs[0]._key, "a");
-        assertTrue(docs[1].error);
-        assertEqual(docs[1].errorNum, ERRORS.ERROR_ARANGO_DOCUMENT_KEY_BAD.code);
-        collection.remove("a");
+        }];
+
+        if (!cluster) {
+          const docs = collection.insert(batch);
+          assertEqual(docs.length, 2);
+          assertEqual(docs[0]._key, "a");
+          assertTrue(docs[1].error);
+          assertEqual(docs[1].errorNum, ERRORS.ERROR_ARANGO_DOCUMENT_KEY_BAD.code);
+          collection.remove("a");
+          assertEqual(collection.count(), 0);
+        } else {
+          try {
+            collection.insert(batch);
```

Comment:
Wouldn't this different test code mean that we establish a new behavior difference between single server and cluster?
An insert should not have different behavior between single server and cluster. When a batch of documents is inserted, the overall insert should **not** fail even if some/all of the operations in the batch fail. Instead, the caller must check the individual results of each operation.